File: clients/python/retab/types/automations/links.py
from typing import Optional

import nanoid  # type: ignore
from pydantic import BaseModel, Field, computed_field

from ..logs import AutomationConfig, UpdateAutomationRequest
from ..pagination import ListMetadata


class Link(AutomationConfig):
    @computed_field
    @property
    def object(self) -> str:
        return "automation.link"

    id: str = Field(default_factory=lambda: "lnk_" + nanoid.generate(), description="Unique identifier for the extraction link")

    # Link Specific Config
    password: Optional[str] = Field(None, description="Password to access the link")


class ListLinks(BaseModel):
    data: list[Link]
    list_metadata: ListMetadata


# Inherits from the methods of UpdateAutomationRequest
class UpdateLinkRequest(UpdateAutomationRequest):
    # ------------------------------
    # Link Config
    # ------------------------------
    password: Optional[str] = None

File: clients/node/src/types/automations/links.ts
import { AutomationConfig, UpdateAutomationRequest } from '../logs.js';
import { ListMetadata } from '../pagination.js';

export interface Link extends AutomationConfig {
  object: 'automation.link';
  id: string;
  password?: string;
}

export interface ListLinks {
  data: Link[];
  list_metadata: ListMetadata;
}

export interface UpdateLinkRequest extends UpdateAutomationRequest {
  password?: string;
}